helix-term/src/ui/editor.rs

```diff
@@ -770,6 +770,69 @@ impl Component for EditorView {
 
                 EventResult::Consumed(callback)
             }
+            Event::Mouse(MouseEvent {
+                kind: MouseEventKind::Down(MouseButton::Left),
+                row,
+                column,
+                modifiers,
+                ..
+            }) => {
+                let editor = &mut cx.editor;
+                let tree = &mut editor.tree;
+                let mut id: Option<ViewId> = None;
+                let mut result = EventResult::Ignored;
+
+                for (view, view_id) in tree.views_with_keys_mut() {
+                    if !view.verify_screen_coords(row as usize, column as usize) {
+                        continue;
+                    }
+
+                    id = Some(view_id);
+
+                    let doc = &mut editor.documents[view.doc];
+                    let jump = (doc.id(), doc.selection(view.id).clone());
+                    view.jumps.push(jump);
+
+                    let pos = view
+                        .pos_at_screen_coords(doc, row as usize, column as usize)
+                        .unwrap();
+
+                    if modifiers == crossterm::event::KeyModifiers::ALT {
+                        let selection = doc.selection(view.id).clone();
+                        doc.set_selection(view.id, selection.push(Range::point(pos)));
+                    } else {
+                        doc.set_selection(view.id, Selection::point(pos));
+                    }
+
+                    result = EventResult::Consumed(None);
+                }
+
+                if let Some(id) = id {
+                    cx.editor.tree.focus = id;
+                }
+
+                result
+            }
+
+            Event::Mouse(MouseEvent {
+                kind: MouseEventKind::Drag(MouseButton::Left),
+                row,
+                column,
+                ..
+            }) => {
+                let (view, doc) = current!(cx.editor);
+
+                let pos = match view.pos_at_screen_coords(doc, row as usize, column as usize) {
+                    Some(pos) => pos,
+                    None => return EventResult::Ignored,
+                };
+
+                let selection = doc.selection(view.id).clone();
+                let primary = selection.primary();
+
+                doc.set_selection(view.id, selection.push(Range::new(primary.anchor, pos)));
```

Comment:
Notice: this is incorrect, I'm working on a fix. This way won't remove old selection while dragged so selection will only grow following the cursor.

Reply:
Instead of pushing you want just edit the primary selection range.